tests/python/test_bit_array_vectorization.py

```diff
@@ -41,3 +41,49 @@ def verify():
     init()
     assign_vectorized()
     verify()
+
+
+@ti.test(require=ti.extension.quant, debug=True, cfg_optimization=False)
+def test_offset_load():
+    ci1 = ti.type_factory_.get_custom_int_type(1, False)
+
+    x = ti.field(dtype=ci1)
+    y = ti.field(dtype=ci1)
+    z = ti.field(dtype=ci1)
+
+    N = 4096
+    n_blocks = 4
+    bits = 32
+    boundary_offset = 1024
+
+    block = ti.root.pointer(ti.ij, (n_blocks, n_blocks))
+    block.dense(ti.ij, (N // n_blocks, N // (bits * n_blocks)))._bit_array(
+        ti.j, bits, num_bits=bits).place(x)
+    block.dense(ti.ij, (N // n_blocks, N // (bits * n_blocks)))._bit_array(
+        ti.j, bits, num_bits=bits).place(y)
+    block.dense(ti.ij, (N // n_blocks, N // (bits * n_blocks)))._bit_array(
+        ti.j, bits, num_bits=bits).place(z)
+
+    @ti.kernel
+    def init():
+        for i, j in ti.ndrange((boundary_offset, N - boundary_offset),
+                               (boundary_offset, N - boundary_offset)):
+            x[i, j] = ti.random(dtype=ti.i32) % 2
+
+    @ti.kernel
+    def assign_vectorized():
+        ti.bit_vectorize(32)
+        for i, j in x:
+            y[i, j] = x[i, j + 1]
+            z[i, j] = x[i, j - 1]
```

Comment:
If we are ready, we can test `x[i - 1, j]/x[i + 1, j]` too. You can also allocate only the `y` field, and reuse the functions `assign_vectorized` and `verify` by adding template arguments `dx: ti.template(), dy: ti.template()`, and the loop bodies should be something like `y[i, j] = x[i + dx, j + dy]`.